test_e2e/dest/couchbase.sh

```diff
@@ -0,0 +1,100 @@
+#!/usr/bin/env bash
+set -e
+
+DEST_PATH=$(dirname "$(realpath "$0")")
+SCRIPT_DIR=$(dirname "$DEST_PATH")
+cd "$SCRIPT_DIR"/.. || exit 1
+OUTPUT_FOLDER_NAME=couchbase-dest
+OUTPUT_ROOT=${OUTPUT_ROOT:-$SCRIPT_DIR}
+OUTPUT_DIR=$OUTPUT_ROOT/structured-output/$OUTPUT_FOLDER_NAME
+WORK_DIR=$OUTPUT_ROOT/workdir/$OUTPUT_FOLDER_NAME
+DOWNLOAD_DIR=$OUTPUT_ROOT/download/$OUTPUT_FOLDER_NAME
+max_processes=${MAX_PROCESSES:=$(python3 -c "import os; print(os.cpu_count())")}
+DESTINATION_CB_SCOPE="_default"
+DESTINATION_CB_COLLECTION="_default"
+CI=${CI:-"false"}
+
+source "$SCRIPT_DIR"/env_setup/couchbase/common/constants.env
+
+# Check if all necessary environment variables are set
+if [ -z "$CB_USERNAME" ] || [ -z "$CB_PASSWORD" ] || [ -z "$CB_CONN_STR" ] || [ -z "$CB_BUCKET" ];  then
+  echo "Error: One or more environment variables are not set. Please set CB_CONN_STR, CB_USERNAME, CB_PASSWORD, and CB_BUCKET"
+  exit 1
+fi
+
+# shellcheck disable=SC1091
+source "$SCRIPT_DIR"/cleanup.sh
+
+function cleanup() {
+
+   # Remove docker container
+  echo "Stopping Couchbase Docker container"
+  docker-compose -f "$SCRIPT_DIR"/env_setup/couchbase/common/docker-compose.yaml down --remove-orphans
+
+  # Kill couchbase background process
+  pgrep -f couchbase-dest | xargs kill
```

Comment:
Why is this needed if couchbase is being run within docker?

Reply:
I actually thought this was some unstructured process. Totally my bad her. Removed this